src/state/slices/opportunitiesSlice/selectors.ts

```diff
@@ -367,7 +371,7 @@ export const selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty =
       // Keep only the version with actual data if it exists, else keep the zero'd out version
       const aggregatedEarnUserStakingOpportunitiesIncludeEmpty = uniqBy(
         [...aggregatedEarnUserStakingOpportunities, ...emptyEarnOpportunitiesTypes],
-        'contractAddress',
+        ({ contractAddress, assetId }) => contractAddress ?? assetId,
```

Comment:
Leveraging the identity nature of `uniqBy` iteratee to either keep the uniques by `contractAddress` (e.g FOX Farming contracts) or by `assetId` for native assets